geode-core/src/main/java/org/apache/geode/cache/query/Index.java

```diff
@@ -115,4 +115,18 @@ default boolean isValid() {
    * @return the projectionAttributes, or "*" if there were none specified at index creation.
    */
   String getCanonicalizedProjectionAttributes();
+
+  /**
+   * Set whether maintenance of index is ongoing.
+   */
+  default void setMaintenanceOngoing(boolean b) {
+    // nothing by default
+  };
+
+  /**
+   * Get whether maintenance of index is ongoing.
+   */
+  default boolean isMaintenanceOngoing() {
```

Comment:
I don't think "maintenance" conveys the right meaning.  IMO it implies that an action invoked by an operator is active.  Something like `isUpToDate()` sounds more like what you want to know.